### DISCUSSION TITLE
Yarn recommendation is outdated

### DISCUSSION THREAD

**Discussion by aleksey-hoffman** [nklayman/vue-cli-plugin-electron-builder#1298] [Q&A] [answered]
Hey @nklayman 
I've noticed that the docs strongly suggest to use Yarn v1, which is pretty much deprecated now and has 0 advantages over NPM v7.

### Consider this:
- Yarn v1 has entered maintenance mode in January 2020 and will eventually reach their end-of-life in terms of support.
- NPM v7 is now as fast as Yarn v1 and has all of its features, plus a great audit tool that shows you all the vulnerabilities during installation process unlike Yarn.
- Why install and learn another dependency (Yarn) that doesn't have any advantages over NPM, which everyone already has installed? It's an unnecessary complication

Perhaps you should consider removing `Yarn v1` recommendation and maybe instead consider `Yarn v2` or `PNPM` as an alternative to `NPM` since it at least provides some benefits (like reduced drive space usage)

**Comment by nklayman** [comment, marked as answer]
According to https://github.com/pnpm/benchmarks-of-javascript-package-managers, yarn is still faster than npm in the most common scenarios. Yarn 2 pnp and pnpm don't play nice with electron native modules afaik (and don't have great vue-cli support either I believe), so I'm going to avoid recommending them. In addition, yarn has some nice features like allowing you to run package bins without having to list them in `scripts`, which I use often. npm has gotten better, however. What do you think about continuing to provide instructions for both tools, but removing the line that recommends yarn over npm?

**Reply by aleksey-hoffman** [reply]
Good to know Yarn still has some advantages in some areas. I wish PNPM would get more traction though, Those node_modules directories are a waste of space.

Yeah I think the instructions for both tools without a strong recommendation is the way to go. Don't quote me on this, but I think there was a similar strong recommendation in the Vue / VueCLI docs, but it looks like it's not there anymore, or perhaps I'm just misremembering this.

Also, I think that package managers might not be the biggest bottleneck in development, it's things like Webpack that waste the most time, it seems. Have you ever considered migrating from Webpack to Evan You's `Vite` packager? I found [this issue](https://github.com/vitejs/vite/issues/911), there's a few links to different Electron-Vite boilerplates there, so it seems like it can be done.

**Reply by nklayman** [reply]
This plugin relies so heavily on vue-cli that it would probably be best to just rewrite it entirely for vite :joy:. Unfortunately I don't have time for that now, maybe I'll get around to it this summer. I also think Vue CLI used to have a recommendation for Yarn, I think that's why I added it to the plugin docs. I'll go ahead and remove it now, though. Thanks for your suggestions!

**Reply by aleksey-hoffman** [reply]
Yeah, I suppose it would take a lot of time, I think it'd be better to stick with Webpack 5 for now and spend the development time on something more important. Things like Tauri, and that.

Thanks, I think this can be marked as answered then 👍

**Reply by nklayman** [reply]
It has to be a Q&A question to be marked as answered. It's in general so I can't mark an answer.

**Reply by nklayman** [reply]
Okay I switched it to Q&A
